shared/tests/data/CoordinateTests.yaml

```diff
@@ -0,0 +1,843 @@
+  - [1.158215718506735e+19, -0.082092487792463e+19, -0.128412669345756e+19]
+  - [-0.221177978857723e+19, -0.587294610603860e+19, 1.435749573211346e+19]
+  - [0.968375936781638e+19, 1.150023295570274e+19, -0.850659770110297e+19]
+  - [-1.812363738774686e+19, 1.099725264800027e+19, -1.384058296612486e+19]
+  - [1.826356284661767e+19, -0.803387818348255e+19, -0.972521326637361e+19]
+  - [-1.567780073893172e+19, -1.061799026989318e+19, -0.896607078176482e+19]
+  - [-1.218955483308987e+19, -0.920506037134477e+19, -1.551252690180505e+19]
+  - [0.052377020200300e+19, -1.057171552663306e+19, 0.627596031673933e+19]
+  - [1.029572129823921e+19, 0.197701275520618e+19, 0.091672504199485e+19]
+  - [-0.521003667178693e+19, 0.094919066251467e+19, 0.678450145642286e+19]
+  - [-0.762391976692548e+19, -0.720011711525928e+19, -0.031443760456174e+19]
+  - [-1.283681015885078e+19, 1.231678934095174e+19, -0.802708543092133e+19]
+  - [1.495217139404973e+19, -1.551102136453661e+19, 0.088992404167258e+19]
+  - [-0.838213197457646e+19, -0.365601213452562e+19, 1.567323558665757e+19]
+  - [-0.583345299333880e+19, 1.475773717182935e+19, 0.933300961684062e+19]
+  - [-1.660598250605305e+19, 1.382319308976448e+19, -0.011619335596907e+19]
+  - [-1.089756578214201e+19, -1.013803514046323e+19, 1.447538814150365e+19]
+  - [0.087384849164657e+19, -0.980341953987881e+19, 1.537155243651548e+19]
+  - [-0.969239446910151e+19, 1.597713050109788e+19, -0.165841504571241e+19]
+  - [0.339525506711963e+19, 0.555164297473728e+19, -0.640815825709116e+19]
+  - [1.154827411151740e+19, -0.810217944931940e+19, -0.602147547703868e+19]
+  - [-0.419008209126276e+19, -1.410252546762864e+19, -0.898108036072751e+19]
+  - [1.043648616776534e+19, 0.769384574341373e+19, -0.977832154324422e+19]
+  - [0.793241160127994e+19, 0.281756063703886e+19, -0.516650790940869e+19]
+  - [1.843752994729693e+19, -1.373740453561332e+19, -0.799107308347852e+19]
+  - [-1.769809607039277e+19, -1.236341043485832e+19, 1.791782767310794e+19]
+  - [0.587220520335672e+19, 0.818822599812079e+19, -0.029066082107191e+19]
+  - [0.121431173783529e+19, -1.742120389560180e+19, 0.703854696814365e+19]
+  - [1.686872888923929e+19, -0.863059466482042e+19, -1.275073913088488e+19]
+  - [0.720689641125352e+19, 1.702906809954598e+19, -0.209191319391396e+19]
+  - [-0.119859553790866e+19, -1.754047274596617e+19, -1.252390030356249e+19]
+  - [-0.750432588559682e+19, 1.091138756493226e+19, -0.363685006307817e+19]
+  - [-0.292361877887738e+19, 1.048582936196728e+19, 0.356833842856448e+19]
+  - [-0.383002588349182e+19, -1.486132196006383e+19, 1.272020870169739e+19]
+  - [-1.463015950215163e+19, 0.790902876388827e+19, 1.508782570074385]
+
+sphericalInput:
+  - [-5, 0, 0]
+  - [-1.4853e-307, 0, 0]
+  - [0, 0, 0]
+  - [1.4853e-307, 0, 0]
+  - [1, 0, 0]
+  - [-5, 1.5707963267948966, 0]
+  - [-5, 0, 1.5707963267948966]
+  - [-5, 1.5707963267948966, 1.5707963267948966]
+  - [-1.4853e-307, 1.5707963267948966, 0]
+  # - [-1.4853e-307, 0, 1.5707963267948966]
```

Comment:
> This is really clean! Nice job.
> 
> Sidenote: We talked a while ago about adding a readme or comment somewhere nearby to test data which detailed how inputs were generated and how the ground truth was generated/checked, so that future people could verify, regenerate, or build upon it. Do you think we could include something like that here?

Yeah for sure. I'll whip something up

Reply:
> 

Yeah something didn't like them. I think it was the yaml cpp. I've completely removed them.